test/Microsoft.Dnx.Testing.CommonUtils/Directory/DirMismatchException.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Xunit.Sdk;
+
+namespace Microsoft.Dnx.Testing
+{
+    public class DirMismatchException : XunitException
+    {
+        public DirMismatchException(string expectedDirPath, string actualDirPath, DirDiff diff)
+        {
+            Message = $@"The actual directory structure '{actualDirPath}' doesn't match expected structure '{expectedDirPath}'.
+Difference information:
+Extra: {string.Join(", ", diff.ExtraEntries)}
```

Comment:
Probably you should join them with a new line no just comma. Maybe new line and a space so it has a little indentation?